### PR TITLE
Release 0.19.10

### LINKED ISSUE #4351: Release `kedro` 0.19.10
### Description

Release Kedro version `0.19.10`

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Resolves #4351 

## Development notes

* Updated Citation, command reference doc, kedro init file
* Updated Release note

## Blocked by
- [x] #4261
- [x] #4265

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [x] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by merelcht** [comment]
https://github.com/kedro-org/kedro-starters/pull/260 also needs to go in before the release!

File: RELEASE.md [hunk 1]
```diff
@@ -20,6 +28,7 @@
 * Added example to explain how dataset factories work.
 
 ## Community contributions
+Many thanks to the following Kedroids for contributing PRs to this release:
```
Comment (astrojuanlu) [inline RELEASE.md:31]: @ravi-kumar-pilla could you double check in the list of merged PRs if we should acknowledge anybody else?

Comment (ravi-kumar-pilla) [reply RELEASE.md:32]: Thank you @astrojuanlu for pointing out. Added other contributions now.

**Review by astrojuanlu — APPROVED** [review]
ship-it

**Review by merelcht — APPROVED** [review]
image

**Pull request merged**